### DISCUSSION TITLE
Facing issue with Event Listener SPI for KC 23.0.5

### DISCUSSION THREAD

**Discussion by Rohit14rr** [keycloak/keycloak#26795] [Q&A] [answered]
I'm trying to write my KC events to a Azure queue using Event Listener SPIs
example - https://github.com/keycloak/keycloak-quickstarts/tree/latest/extension/event-listener-sysout
I've created a jar with my dependencies and dropped it in KC provider folder.
But KC fails to start and gives error : 
"Caused by: java.lang.LinkageError: loader constraint violation: loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @16f7b4af wants to load class org.keycloak.common.Profile$ProfileName. A different class with the same name was previously loaded by java.net.URLClassLoader @1b266842."
Am I missing something here or not following the example correctly.
I'm using jdk v17.
[pom.txt](https://github.com/keycloak/keycloak/files/14170644/pom.txt)

**Comment by danielFesenmeyer** [comment]
Maybe the keycloak version defined in your pom and the version of your keycloak server are different?

In any case, it seems like - in contrast to your other keycloak dependencies - the keycloak-core dependency is NOT defined as provided. This way your "fat jar" gets unnecessarily big and may cause such issues.

**Reply by danielFesenmeyer** [reply, marked as answer]
If not defined as provided, it will be handled as a compile dependency by the maven-assembly-plugin, which means it will be included into the "fat jar".
